Sources/_StringProcessing/Engine/MEBuilder.swift

```diff
@@ -153,33 +153,61 @@ extension MEProgram.Builder {
   mutating func buildAdvance(_ n: Distance) {
     instructions.append(.init(.advance, .init(distance: n)))
   }
-  
+
+  mutating func buildReverse(_ n: Distance) {
+    instructions.append(.init(.reverse, .init(distance: n)))
+  }
+
+  mutating func buildReverseUnicodeScalar(_ n: Distance) {
+    instructions.append(.init(.reverse, .init(distance: n, isScalarDistance: true)))
+  }
+
   mutating func buildAdvanceUnicodeScalar(_ n: Distance) {
     instructions.append(
       .init(.advance, .init(distance: n, isScalarDistance: true)))
   }
-  
+
   mutating func buildConsumeNonNewline() {
     instructions.append(.init(.matchAnyNonNewline, .init(isScalar: false)))
   }
-                        
+
+  mutating func buildReverseConsumeNonNewline() {
```

Comment:
As a thought, would it makes sense to have a required `reverse: Bool` parameter instead of a separate function? This would mean that every call site has to think about whether it is reversing or not, which could help avoid bugs from forgetting to check for reverse. If every call site needs to think about (or pass along) a reverse value, that would also clean up the branches there.

Reply:
Yeah I think that would drastically cut down on duplicated code as well.